rust/src/format/manifest.rs

```diff
@@ -79,6 +80,16 @@ impl Manifest {
         )
     }
 
+    /// Set the `timestamp_nanos` value from a Utc DateTime
+    pub fn set_timestamp(&mut self, timestamp: Option<SystemTime>) {
+        let timestamp = timestamp.unwrap_or_else(SystemTime::now);
+        let nanos = timestamp
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .unwrap()
+            .as_nanos();
+        self.timestamp_nanos = nanos as u128;
```

Comment:
Is this persisted in protobuf? 
do we need `nano` resolution ?

Also might want to have a timezone, so that this is reliable for either "system clock" or cross-datacenter writes.

Reply:
> Is this persisted in protobuf?

Yes. This is just the existing behavior moved over to this file so it's easier to work with.

But I'll note this in our doc on OCC, since we might want to update when considering concurrent writers.